analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionEvaluatesToConstant.cs

```diff
@@ -2720,3 +2720,24 @@ void LocalFunction()
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/3955 for S2583
+namespace Repro_RefParam
```

Comment:
I think this test cases is similar to the one in the `ConditionEvaluatesToConstant.CSharp8.cs ` file - see https://github.com/SonarSource/sonar-dotnet/pull/3358/files#diff-1b8068370be2d846e5ffe4e1592e213810d5b148f949f60deca575648ac66e6cR230

Although I like this more, as it's more readable.

Reply:
or maybe the syntax is slightly different there?

Reply:
my suggestion is to:
- remove the other example from the CSharp8 test case, as it's not necessarily C# 8 syntax
-  as I said, I like you example more because it's simpler. I'd delete the other one, and update the link in your sample to point to the original issue.
- in the original issue thread, also add this snippet so that we can see the simple syntax directly from the issue , without going to the test case

Reply:
and mention here that the problem is in a multi-threaded environment, where the `stop` can be modified in a different thread while this thread is running